ibc-clients/ics07-tendermint/src/client_state/update_client.rs

```diff
@@ -10,219 +10,228 @@ use tendermint_light_client_verifier::Verifier;
 
 use super::ClientState;
 use crate::consensus_state::ConsensusState as TmConsensusState;
-use crate::context::{CommonContext, ValidationContext as TmValidationContext};
-
-impl ClientState {
-    pub fn verify_header<ClientValidationContext>(
-        &self,
-        ctx: &ClientValidationContext,
-        client_id: &ClientId,
-        header: TmHeader,
-    ) -> Result<(), ClientError>
-    where
-        ClientValidationContext: TmValidationContext,
+use crate::context::{CommonContext, TmVerifier, ValidationContext as TmValidationContext};
+
+pub fn verify_header<ClientValidationContext>(
+    client_state: &ClientState,
+    ctx: &ClientValidationContext,
+    client_id: &ClientId,
+    header: TmHeader,
```

Comment:
a small nit: let's take the reference of `TmHeader` here since the ownership is not needed.